modules/KalturaSupport/components/dualScreen/dualScreen.js

```diff
@@ -621,7 +637,133 @@
 					at: location[0]+location[1],
 					of: $( this.getPlayer().getInterface() )
 				});
-			}
+			},
+			getKClient: function () {
+				if (!this.kClient) {
+					this.kClient = mw.kApiGetPartnerClient(this.embedPlayer.kwidgetid);
+				}
+				return this.kClient;
+			},
+			/**
+			 * Searches for the first/next cue point after the current player time
+			 * @returns a matching cue point if found, null otherwise
+			 */
+			getCurrentPlayerViewModeCuePoint: function ( ) {
+
+				var currentTime = this.getPlayer().currentTime *1000;
+				var cuePoints = this.getPlayerViewModeCuePoints();
+				var cuePoint;
+				var duration=this.getPlayer().isLive() ? 0 : this.getPlayer().getDuration() * 1000;
+
+				//assume sortedCuePoints array
+				for ( var i = 0; i < cuePoints.length; i++ ) {
+
+					var startTime = cuePoints[i].startTime;
+
+					if ( (startTime > currentTime) ||  //stop once we found a future slide (or out of range slide)
+						(duration>0 && startTime>duration)) {
+						break;
+					}
+
+					cuePoint=cuePoints[i];
+
+				}
+
+				// NOTE: the code below can assist debugging issues with play view mode sync
+				// But it will polute the log console so keep it commented unless required for debugging purposes
+				//if (cuePoint)
+				//{
+				//	var nextCuePoint = i < cuePoints.length ? cuePoints[i] : null;
+				//	var message= 'dualscreen: current cue point (' + cuePoint.id + ') time ' + new Date(cuePoint.startTime) + ' | server time ' + new Date(currentTime);
+                //
+				//	if (nextCuePoint) message += ' | next cue point (' + nextCuePoint.id + ') time ' + new Date(nextCuePoint.startTime);
+                //
+				//	mw.log(message);
+				//}
+				return cuePoint;
+			},
+			/**
+			 * Gets all the cue points of type player view mode
+			 * @returns {Array}
+			 */
+			getPlayerViewModeCuePoints : function()
+			{
+				var _this = this;
+				var cuePoints = [];
+				if ( this.getPlayer().kCuePoints ) {
+					cuePoints = _this.getPlayer().kCuePoints.getCodeCuePointsByTag({tag : 'player-view-mode', sortDesc : false});
```

Comment:
@OrenMe I agree it is expensive, but since the dualscreen plugin is very complex we decided to use the same patterns/disciplines you are using to create a similiar flow with what we currently have.  

A. in file ['mw.kCuePoints.js' line 348](https://github.com/kaltura/mwEmbed/blob/webc-679/modules/KalturaSupport/resources/mw.KCuePoints.js#L348) you invoke function 'getNextCuePoint' every 250ms which does the same behavior. 
B. In file ['imageplayer.js' line 63](https://github.com/kaltura/mwEmbed/blob/webc-679/modules/KalturaSupport/components/dualScreen/imagePlayer.js#L63) if the cuepoint is not the one you are looking for you also invoke a loop against the list of cuepoints.

I can do enhancements to that logic like using a **priority queue** per cuepoint type which eliminate the need for re-looping the list every time, but I think this is something that the infrastructure should have handled. In order to **not affect** current image player behavior we decided to mimic the same behavior as done for the thumbs.

@OrenMe  @kalturaguy what do you think? Should I do the improvement for my part (only) but then the thumb code will still be doing the re-loop discipline and we will have different behavior in code?